plenum/server/replica.py

```diff
@@ -490,12 +490,8 @@ def on_view_change_done(self):
 
     def on_propagate_primary_done(self):
         assert self.isMaster
-        # if this is a Primary that is re-connected (that is view change is not actually changed,
-        # we just propagate it, then make sure that we don;t break the sequence
-        # of ppSeqNo
-        if self.isPrimary:
-            self.lastPrePrepareSeqNo = self.last_ordered_3pc[1]
-            self.update_watermark_from_3pc()
+        self.lastPrePrepareSeqNo = self.last_ordered_3pc[1]
```

Comment:
Please add a test for non-primaries (there is one `test_recover_stop_primaries`, but it's for primaries only)